clients/spark/core/src/test/scala/io/treeverse/utils/AsInterfaceTest.scala

```diff
@@ -0,0 +1,71 @@
+package io.treeverse.utils
+
+import org.scalatest.funsuite.AnyFunSuite
+
+import org.objectweb.asm.{ClassReader, ClassWriter}
+import org.objectweb.asm.util.CheckClassAdapter
+
+import java.io.ByteArrayOutputStream
+import com.google.common.io.ByteStreams
+
+class AsInterfaceTest extends AnyFunSuite {
+  test("[internal] Boo does not implement Foo") {
+    assertThrows[ClassCastException] {
+      (new Boo(9)).asInstanceOf[Foo]
+    }
+  }
+
+  test("AsInterface: Boo as Foo") {
+    val booStream = this.getClass.getClassLoader.getResourceAsStream("io/treeverse/utils/Boo.class")
+    val booBytes = new ByteArrayOutputStream
+    ByteStreams.copy(booStream, booBytes)
+
+    val cr = new ClassReader(booBytes.toByteArray)
+    val cw = new ClassWriter(cr, 0)
+    val cc = new CheckClassAdapter(cw);
+    val ai = new AsInterface(cc, Map(("Foo", classOf[Foo])))
+    cr.accept(ai, 0)
```

Comment:
Why are you repeating the logic of `AsInterfaceClassLoader` in a test?

Reply:
This tests `AsInterface` on its own, and is also a "simple" example of how `AsInterface` works.  Dunno, if you don't like it I shall remove it.